backend/DB/api/static/graph_display.js

```diff
@@ -0,0 +1,177 @@
+let width = 1500, height = 750;
+
+let radius = width / 50;
+
+d3.json("api/user/all", (users) => {
+
+    d3.json("api/pair/count_pair", (edge_users) => {
+
+        d3.json("status/data", (status) => {
+
+            let cake = 20, pizza = 40;
+            let cake_thres = 30, pizza_thres = 70;
+            let cake_claim = 2, pizza_claim = 1;
+
+            console.log(status);
+
+            let user_indices = {};
+            users.forEach((v, i) => user_indices[v.username] = i);
+            let links = edge_users.map((v) => ({
+                'source': user_indices[v.source],
+                'target': user_indices[v.target],
+                'total': v.total,
+            }));
+
+
+            let setCakeProgress = d3.select('#cake')
+                .attr('style', "width:" + status.cake_count / status.cake_thres * 100 + "%")
+                .attr("aria-valuemax", status.cake_thres)
+                .text(status.cake_count.toString() + "/" + status.cake_thres.toString());
+
+            let setPizzaProgress = d3.select('#pizza')
+                .attr('style', "width:" + status.pizza_count / status.pizza_thres * 100 + "%")
+                .attr("aria-valuemax", status.pizza_thres)
+                .text(status.pizza_count.toString() + "/" + status.pizza_thres.toString());
+
+            //Add patterns to images
+            let defs = d3.select('#patterns_svg')
+                .selectAll('pattern')
+                .data(users)
+                .enter()
+                .append('pattern')
+                .attr('id', function (d) {
+                    return d.username
+                })
+                .attr('height', "100%")
+                .attr('width', "100%")
+                .attr('patternContentUnits', 'objectBoundingBox')
+                .append('image')
+                .attr('height', 1)
+                .attr('width', 1)
+                .attr('preserveAspectRatio', 'none')
+                .attr('href', function (d) {
+                    if (d.image === "unknown") {
+                        return "../static/default.png"
+                    } else {
+                        return d.image
+                    }
+                });
+
+            //Add nodes to <class>-es
+            let node = d3.select('#circles')
+                .selectAll('circle')
+                .data(users)
+                .enter().append('circle')
+                .attr('class', 'user_node')
+                .attr('r', radius)
+                .attr('fill', function (d) {
```

Comment:
Arrow function. Since this function is just a single expression, you do not need curly braces or the return keyword (as long as it is an arrow function).